src/librustc_lexer/src/unescape.rs

```diff
@@ -243,6 +289,8 @@ where
                 let second_char = chars.clone().next();
                 match second_char {
                     Some('\n') => {
+                        // It's likely to have a whitespace sequence after newline
+                        // so we can try to skip them.
```

Comment:
We do this not because it's some kind of optimization, but because the langauge definition requires that we skip an **escaped** newline and the following whitespace.